### DISCUSSION TITLE
How to set label and sort on $action->getRecordSelect()

### DISCUSSION THREAD

**Discussion by billydekid** [filamentphp/filament#7241] [Help]
Hi,

Is it possible to set label and sort default on $action->getRecordSelect() custom form?

Thanks,
[bayu]

**Comment by leandrocfe** [comment]
Not sure about the sort, but did you try `->label('new label')` ?

**Reply by billydekid** [reply]
I've tried it before and it hasn't worked.

**Comment by pboivin** [comment]
@billydekid For the sort, you can hook into the query with `recordSelectOptionsQuery()`:
https://filamentphp.com/docs/2.x/admin/resources/relation-managers#scoping-the-options

**Reply by billydekid** [reply]
@pboivin tried with this but still not work: 

` $action->getRecordSelect()->recordSelectOptionsQuery(fn (Builder $query) => $query->orderBy('id', 'asc')),
`

with an error msg:

BadMethodCallException
Method Filament\Forms\Components\Select::recordSelectOptionsQuery does not exist.

**Reply by pboivin** [reply]
Try it on the $action

**Reply by billydekid** [reply]
Thanks @pboivin

**Comment by sumardi** [comment]
```php
Tables\Actions\AttachAction::make()
    ->recordSelect(fn ($select) => $select
        ->hiddenLabel(false) // v3
        ->disableLabel(false) // v2
        ->label(__('My Label'))
    )
    ->recordSelectOptionsQuery(fn ($query) => $query->orderBy('name', 'asc'))
```

`hiddenLabel()` in v3, `disableLabel()` in v2

**Reply by billydekid** [reply]
@sumardi , the label is work but the sort was not as expected.
I tried to debug by passing an invalid column name and I could view the actual query which has a first and default sort/order statement: `"education_level" ASC`, instead of "id" ordered. How to drop this default order statement?

```
SELECT
  *
FROM
  "education"
ORDER BY
  "education_level" ASC,
  "xyz123" ASC
```
Following are the RelationManager file and table structure.

EducationRelationsManager.php
-----

```php
            ->headerActions([
                Tables\Actions\AttachAction::make()
                    ->preloadRecordSelect()
                    ->color('primary')
                    ->label('New Education')
                    ->form(fn (AttachAction $action): array => [
                        // apply Card layout on modal. my question is answered by danharrin
                        // https://github.com/filamentphp/filament/discussions/6495#discussioncomment-5913137
                        Card::make([
                            $action->getRecordSelect(),
                            Forms\Components\TextInput::make('education_name')
                                ->label('School/Institute name')
                                ->required(),
                            Forms\Components\TextInput::make('major')->required(),
                            Forms\Components\TextInput::make('score')->required(),
                            Forms\Components\TextInput::make('start_year')->numeric()->maxLength(4)->required(),
                            Forms\Components\TextInput::make('end_year')->numeric()->maxLength(4)->required(),
                        ])
                            ->columns(2)
                    ])
                    ->recordSelect(
                        fn ($select) => $select
                            ->disableLabel(false) // v2
                            ->label(__('Education Level'))
                    )
                    ->recordSelectOptionsQuery(fn ($query) => $query->orderBy('id', 'asc'))
                    ->modalHeading('New Education'),
            ])
```

employees
--------------
id
employee_number
user_id
full_name
gender
marital_status

educations
--------------
id
education_level
description
created_at
updated_at

education_employee (intermediate table)
--------------
id
employee_id
education_id
education_name
major
score
start_year
end_year
created_at
updated_at

Education model
--------------------
```php
    public function employees(): BelongsToMany
    {
        return $this->belongsToMany(Employee::class)
            ->withPivot('id', 'education_id', 'education_name', 'major', 'score', 'start_year', 'end_year')
            ->withTimestamps();
    }
```


Employee model
--------------------
```php
    public function educations(): BelongsToMany
    {
        return $this->belongsToMany(Education::class)
            ->withPivot('id', 'education_id', 'education_name', 'major', 'score', 'start_year', 'end_year')
            ->withTimestamps();
    }
```

**Reply by sumardi** [reply]
Remove `id` in the pivot or use `$query->orderBy('educations.id', 'asc')`

**Reply by billydekid** [reply]
I can't remove `id` since I need it to handle allow duplicates [1]
Put the prefix table name still doesn't work either.

[1] https://filamentphp.com/docs/2.x/admin/resources/relation-managers#handling-duplicates
